src/App.tsx

```diff
@@ -17,18 +69,27 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onQueryChange={handleAllTodos}
+                onClear={handleClear}
+                query={query}
+                onStatusChange={handleStatusChange}
+                status={status}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loadingTodos && <Loader />}
+              <TodoList
+                todos={filteredTodosByStatus}
+                onShowModal={handleShowModal}
```

Comment:
```suggestion
                onShowModal={setSelectedTodo}
```
`handleShowModal` function wrapper is redundant

Explanation: `onShowModal` prop expects a function that accepts `todo` as an argument. `setSelectedTodo` is such a function. Also, the result will be the same, because `handleShowModal` does nothing but just call `setSelectedTodo` inside